packages/webpack/lib/setEnv.js

```diff
@@ -0,0 +1,37 @@
+module.exports = setEnv
+
+/**
+ * Replaces constants in your source code with values from `process.env`
+ * using the `webpack.EnvironmentPlugin`.
+ *
+ * Using `setEnv` multiple times results in a single
+ * EnvironmentPlugin instance configured to do all the replacements.
+ *
+ * @param {string[]|object} constants
+ * @return {Function}
+ */
+function setEnv (constants) {
+  const setter = context => prevConfig => {
+    context.setEnv = Object.assign({}, context.setEnv, toObject(constants))
+    return prevConfig
+  }
+
+  return Object.assign(setter, { post: addEnvironmentPlugin })
+}
+
+function addEnvironmentPlugin (context, util) {
+  return util.addPlugin(
+    new context.webpack.EnvironmentPlugin(context.setEnv)
+  )
+}
+
+function toObject (constants) {
+  if (Array.isArray(constants)) {
+    return constants.reduce((acc, c) => {
+      acc[c] = null
```

Comment:
Setting to `null` according to [EnvironmentPlugin docs](https://webpack.js.org/plugins/environment-plugin/).

> To specify an unset default value, use `null` instead of `undefined`.

Reply:
Maybe we should add this note as a comment.

Reply:
Setting to `null` silences [this warning](https://github.com/webpack/webpack/blob/master/lib/EnvironmentPlugin.js#L30). Not sure if it's good.

Reply:
@andywer @sapegin WDYT? It should be `undefined` instead of `null` imo.

Reply:
Shouldn't it rather be `acc[c] = process.env[c]`?

Reply:
Btw, `acc` and `c` are not very good identifiers 😉

Reply:
The [EnvironmentPlugin does it](https://github.com/webpack/webpack/blob/master/lib/EnvironmentPlugin.js#L26) so it makes no sense to do that.

`acc` is for accumulator and kind of standard, `c` is easy to understand from the code. But no problem, I will replace them with `result` and `constant`.

Reply:
> Shouldn't it rather be acc[c] = process.env[c]?

We’re setting default values here, and EnvironmentPlugin will use them these keys are not defined in `process.env`. It’s a bit hard to understand from their docs ;-)